### PR TITLE
chore: Bump flask libs

### PR DESCRIPTION
**Pull request by EugeneTorap**
### SUMMARY
The PR bump the next flask libs:

- flask to 2.1.3
- flask-compress to 1.13
- flask-login to 0.6.0
- flask-talisman to 1.0.0
- flask-wtf to 1.0.1
- werkzeug to 2.1.2

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -80,23 +82,25 @@ flask-appbuilder==4.1.6
...
     # via apache-superset
 flask-sqlalchemy==2.5.1
     # via
     #   flask-appbuilder
     #   flask-migrate
-flask-talisman==0.8.1
+flask-talisman==1.0.0
     # via apache-superset
-flask-wtf==0.14.3
+flask-wtf==1.0.1
```
Comment (dpgaspar) [inline requirements/base.txt:103]: this is a major bump, isn't out of scope given the description?

File: requirements/base.txt [hunk 2]
```diff
@@ -80,23 +82,25 @@ flask-appbuilder==4.1.6
     # via apache-superset
 flask-babel==1.0.0
     # via flask-appbuilder
-flask-caching==1.10.1
+flask-caching==2.0.1
```
Comment (dpgaspar) [inline requirements/base.txt:85]: I think we need to be careful about this one, I'll double check

File: requirements/base.txt [hunk 3]
```diff
@@ -80,23 +82,25 @@ flask-appbuilder==4.1.6
...
     # via apache-superset
-flask-compress==1.10.1
+flask-compress==1.13
     # via apache-superset
 flask-jwt-extended==4.3.1
     # via flask-appbuilder
-flask-login==0.4.1
-    # via flask-appbuilder
-flask-migrate==3.1.0
+flask-login==0.6.2
```
Comment (dpgaspar) [inline requirements/base.txt:91]: this is a major bump, isn't out of scope given the description?

Comment (EugeneTorap) [reply requirements/base.txt:91]: @dpgaspar I had a error with the new flask and old `flask-login` and I bumped it

File: tests/integration_tests/core_tests.py [hunk 4]
```diff
@@ -626,7 +626,7 @@ def test_redirect_invalid(self):
 
         self.login(username="admin")
         response = self.client.get(f"/r/{model_url.id}")
-        assert response.headers["Location"] == "http://localhost/"
+        assert response.headers["Location"] == "/"
```
Comment (EugeneTorap) [inline tests/integration_tests/core_tests.py:629]: > Response.autocorrect_location_header is disabled by default. The Location header URL will remain relative, and exclude the scheme and domain, by default. https://github.com/pallets/werkzeug/issues/2352

> When Response.autocorrect_location_header was added in 2011 https://github.com/pallets/werkzeug/commit/0ec643b1cb48dc5ccef38a569941774d4bba52cd, it was documented as "correct the location header to be RFC conformant". It was referring to [RFC 2616](http://tools.ietf.org/html/rfc2616#section-14.30). That was superseded by [RFC 7231](https://httpwg.org/specs/rfc7231.html#header.location) in 2014, which allows relative URLs. [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location) lists all browsers as compliant with this. Switch autocorrect_location_header to be disabled by default.

File: tests/integration_tests/datasource_tests.py [hunk 5]
```diff
@@ -466,9 +466,9 @@ def test_get_samples(test_client, login_as_admin, virtual_dataset):
         f"/datasource/samples?datasource_id={virtual_dataset.id}&datasource_type=table"
     )
     # feeds data
-    test_client.post(uri)
+    test_client.post(uri, json={})
     # get from cache
-    rv = test_client.post(uri)
+    rv = test_client.post(uri, json={})
```
Comment (EugeneTorap) [inline tests/integration_tests/datasource_tests.py:471]: > Request.get_json() will raise a 400 BadRequest error if the Content-Type header is not application/json. This makes a very common source of confusion more visible. https://github.com/pallets/werkzeug/issues/2339

If there's no json param or `json=None` then 400 http status will be thrown

**Comment by dpgaspar** [comment]
Seeing that https://github.com/advisories/GHSA-7wxw-4483-3m34 is a disputed CVE, and only occurs when running Flask with the provided dev server, something you should never do on production for several reasons.

regarding flask-caching I don't see on the CVE any submitted patches, https://github.com/advisories/GHSA-656c-6cxf-hvcv. This CVE is disputed also, since an attacked would have to be able to write to the cache itself, something that would cause severe issues all around not just by deserializing content with Pickle.

Can you please just bump Pillow and resolve the current PR conflicts? Thank you

**Comment by EugeneTorap** [comment]
@dpgaspar I've created a separate PR for Pillow - #22489
This PR will be for updating flask libs.

File: tests/integration_tests/conftest.py [hunk 6]
```diff
@@ -188,7 +188,11 @@ def remove(self) -> None:
 
 
 def setup_presto_if_needed():
-    backend = app.config["SQLALCHEMY_EXAMPLES_URI"].split("://")[0]
+    db_uri = (
+        app.config.get("SQLALCHEMY_EXAMPLES_URI")
+        or app.config["SQLALCHEMY_DATABASE_URI"]
+    )
+    backend = db_uri.split("://")[0]
```
Comment (dpgaspar) [inline tests/integration_tests/conftest.py:195]: this seems like an unrelated change, no?

File: tests/integration_tests/datasource_tests.py [hunk 7]
```diff
@@ -543,7 +543,7 @@ def test_get_samples_with_filters(test_client, login_as_admin, virtual_dataset):
         f"/datasource/samples?datasource_id={virtual_dataset.id}&datasource_type=table"
     )
     rv = test_client.post(uri, json=None)
-    assert rv.status_code == 200
+    assert rv.status_code == 400
```
Comment (dpgaspar) [inline tests/integration_tests/datasource_tests.py:546]: why did this changed to a 400 now?

**Review by dpgaspar — COMMENTED** [review]
looks good, final comments

File: tests/integration_tests/datasource_tests.py (hunk 7 above)
Comment (EugeneTorap) [reply tests/integration_tests/datasource_tests.py:546]: Werkzeug 2.1.0 changelog:

> Request.get_json() will raise a 400 BadRequest error if the Content-Type header is not application/json. This makes a very common source of confusion more visible. https://github.com/pallets/werkzeug/issues/2339

If there's no json param or `json=None` then 400 http status will be thrown

File: tests/integration_tests/conftest.py (hunk 6 above)
Comment (EugeneTorap) [reply tests/integration_tests/conftest.py:195]: It's the related change. When you want to run local test with `SQLALCHEMY_EXAMPLES_URI = None` you will have an exception here and I fix the issue.

> URI to database storing the example data, points to
SQLALCHEMY_DATABASE_URI by default if set to `None`

**Comment by EugeneTorap** [comment]
@dpgaspar @villebro I have fixed that 400 'Bad Request' problem for datasource/samples [POST] API.
Can you take a look and I guess we can merge the PR.

**Review by dpgaspar — APPROVED** [review]

**Comment by dpgaspar** [comment]
Great work! @EugeneTorap

**Comment by cwegener** [comment]
@EugeneTorap @dpgaspar Seeing that the compiled `.txt` lock files are manually edited in this change as well, is it correct to assume that `pip-compile-multi` is now deprecated and no longer used? Will it be replaced with an alternative any time soon?

**Comment by EugeneTorap** [comment]
Hi @cwegener! I want to prepare a PR with [poetry](https://python-poetry.org/) instead of `pip-compile-multi` usage for dependency management.

**Comment by villebro** [comment]
@cwegener @EugeneTorap the lock files should preferably be updated via `pip-compile-multi`, and IMO it does work quite ok. Having said that, I'd love to see us move over to `poetry`, but I think it's a pretty big task and requires proper coordination to avoid unnecessary regressions or general back and forth.

**Comment by cwegener** [comment]
I find that `poetry`'s dependency management is not quite reliable enough for large codebases.

Anyway. Sounds like there is a need for either discussion of the Python dependency management options, or a better checking of consistent `pip-compile-multi` usage.

**Comment by villebro** [comment]
> I find that `poetry`'s dependency management is not quite reliable enough for large codebases.
> 
> Anyway. Sounds like there is a need for either discussion of the Python dependency management options, or a better checking of consistent `pip-compile-multi` usage.

We should probably add a check to to our linting workflow ensure the compiled lock files are correct. Should be easy.
